src/ifcopenshell-python/ifcopenshell/type_checker.py

```diff
@@ -0,0 +1,149 @@
+from lark import Lark, Transformer
+
+# Grammar to retrieve the following structure: YYYY-MM-DDThh:mm:ss
+grammar = """
+    start: time_value
+    time_value: year "-" month "-" day "T" hour ":" minute ":" second ["." millisecond]
+
+    hour: DIGIT DIGIT
+    minute: DIGIT DIGIT
+    second: DIGIT DIGIT
+    millisecond: DIGIT+
+
+    day: DIGIT DIGIT
+    month: DIGIT DIGIT
+    year: MINUS? DIGIT+
```

Comment:
Wikipedia says: 

> ISO 8601 prescribes, as a minimum, a four-digit year

Should we add that to the grammar?

Reply:
added https://github.com/IfcOpenShell/IfcOpenShell/pull/3864/commits/4a2f34a600a0326946042a6b6f2f302947943601